extensions/gcp/tests/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+include(GoogleTest)
+include(FetchContent)
+FetchContent_Declare(
+        googletest
+        URL            https://github.com/google/googletest/archive/refs/tags/release-1.11.0.tar.gz
+        URL_HASH       SHA256=b4870bf121ff7795ba20d20bcdd8627b8e088f2d1dab299a031c1034eddc93d5
+)
+set(BUILD_GMOCK ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+file(GLOB GCS_TESTS  "*.cpp")
+add_executable(gcp_tests ${GCS_TESTS})
```

Comment:
I think we should aim to follow other test programs, and create one for each source file, unless there is some strong reason not to do so

Reply:
Didnt want to create n+1 main functions but since there is no need for that with gtest_main.
I changed it to be more inline with the other tests. https://github.com/apache/nifi-minifi-cpp/pull/1268/commits/57404afcf541daa1780365e054a555dcd8093c3e#diff-0da9417807df8fa9f894ba2ae5113dfc2fc236f5f966b3f39b80f45b0442eed2R30